docker/worker/worker.py

```diff
@@ -581,7 +581,8 @@ def _log_task_latency(self, message):
       request_time = int(request_time)
       latency = int(time.time()) - request_time
       task_type = message.attributes['type']
-      source_id = get_source_id(message)
+      source_id = get_source_id(message) or message.attributes.get(
```

Comment:
Do you need to do this at 569 as well? it seems to be a similar log

Reply:
Good catch, I think you're right, but this auto-merged before I could address, see #1420 1420